src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -106,6 +114,23 @@ export class DataSourceSelectable extends React.Component<
             ...this.state,
             dataSourceOptions,
           });
+
+          const selectedDataSource = getDefaultDataSource(
+            filteredDataSources,
+            LocalCluster,
+            this.props.uiSettings,
+            this.props.hideLocalCluster,
+            this.props.selectedOption
+          );
+          if (selectedDataSource.length === 0) {
+            this.props.notifications.addWarning('No connected data source available.');
+          } else {
+            this.setState({
```

Comment:
I think `this.setState` here will override `dataSourceOptions` set on line 113 because state update is async, `this.state` here is still the old one, thus `dataSourceOptions` been override.

Reply:
Thanks for pointing this out! Let me chek on this.

Reply:
Found this article on it as "React may batch multiple setState() calls into a single update for performance " in https://legacy.reactjs.org/docs/state-and-lifecycle.html. I am looking into this see how to fix this issue